### PR TITLE
Batch 6: Pretty Momma

### PR DESCRIPTION
**Pull request by ericsorenson**
- 8707 [Court: Content Changes for SPTO/Pretrial Memo - Chambers Phone # - Release with 8662 and 8663](https://github.com/flexion/ef-cms/issues/8707) 
- 8684 [Court: New Document Needed To Support Remote Proceedings](https://github.com/flexion/ef-cms/issues/8684) 
- 8663 [Court: Content Changes for SPTO for Small Cases - Release with 8662](https://github.com/flexion/ef-cms/issues/8663) 
- 8662 [Court: Content Changes for SPTO (Regular Cases) - Release with 8663](https://github.com/flexion/ef-cms/issues/8662) 
- 8652 [Court: Content Changes Needed to Notice of Receipt of Petition](https://github.com/flexion/ef-cms/issues/8652) 
- 8651 [Order Search: Special Character Usage ](https://github.com/flexion/ef-cms/issues/8651) 
- 8650 [Order Search: Case Caption and Docket# section of DAWSON-generated/created content should be searchable content](https://github.com/flexion/ef-cms/issues/8650) 
- 8553 [Usability Testing: Order Search Round ustaxcourt/ef-cms#6492](https://github.com/flexion/ef-cms/issues/8553)

### CODE REVIEW

File: shared/jest.config.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ module.exports = {
     '!src/**/*_.js',
     '!src/persistence/sqs/deleteMessage.js',
     '!src/persistence/sqs/getMessages.js',
+    '!src/persistence/dynamo/**/*.js',
```
Comment (adunkman) [inline shared/jest.config.js]: Could you say more about why this should be excluded from coverage? I see a number of files in this folder which are already covered by tests, which would indicate to me that test coverage is desirable here.

File: shared/src/business/useCases/updatePetitionerInformationInteractor.getIsUserAuthorized.test.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+          ...mockUser,
+          role: ROLES.privatePractitioner,
+          userId: 'a003e912-7b2f-4d2f-bf00-b99ec0d29de1',
+        },
+      });
+
+      expect(isUserAuthorized).toBeFalsy();
+    });
+
+    it('should return false when the user is a petitioner attempting to modify another petitioner', async () => {
```
Comment (adunkman) [inline shared/src/business/useCases/updatePetitionerInformationInteractor.getIsUserAuthorized.test.js]:
```suggestion
    it('should return false when the user is a petitioner attempting to modify another petitioner', () => {
```

File: shared/src/business/useCases/updatePetitionerInformationInteractor.getIsUserAuthorized.test.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+          ...mockUser,
+          role: ROLES.petitioner,
+          userId: 'a003e912-7b2f-4d2f-bf00-b99ec0d29de1',
+        },
+      });
+
+      expect(isUserAuthorized).toBeFalsy();
+    });
+
+    it('should return true when the user is a petitioner its own contact information', async () => {
```
Comment (adunkman) [inline shared/src/business/useCases/updatePetitionerInformationInteractor.getIsUserAuthorized.test.js]:
```suggestion
    it('should return true when the user is a petitioner its own contact information', () => {
```

File: shared/src/business/useCases/updatePetitionerInformationInteractor.getIsUserAuthorized.test.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+          ...mockUser,
+          role: ROLES.petitioner,
+          userId: SECONDARY_CONTACT_ID,
+        },
+      });
+
+      expect(isUserAuthorized).toBeTruthy();
+    });
+
+    it('should return true when the user is representingCounsel', async () => {
```
Comment (adunkman) [inline shared/src/business/useCases/updatePetitionerInformationInteractor.getIsUserAuthorized.test.js]:
```suggestion
    it('should return true when the user is representingCounsel', () => {
```

File: web-api/migration-terraform/main/lambdas/migration-segments.js [hunk 5]
```diff
@@ -165,6 +153,7 @@ exports.handler = async event => {
     ...(await hasMigrationRan('bug-0036-public-served-parties-code.js')),
     ...(await hasMigrationRan('0036-phone-number-format.js')),
```
Comment (adunkman) [inline web-api/migration-terraform/main/lambdas/migration-segments.js:174]: Hmm, I see that the code to actually run the migrations was removed above — should these still be here?

File: web-api/src/v2/v2ApiWrapper.test.js [hunk 6]
```diff
@@ -20,32 +18,31 @@ describe('v2ApiWrapper', () => {
...
   });
+  [401, 403, 404, 500].forEach(statusCode =>
+    it(`error thrown preserves status code ${statusCode}`, () =>
+      expect(() => v2ApiWrapper(throwWithStatus(statusCode))).rejects.toThrow(
+        expect.objectContaining({ statusCode }),
+      )),
+  );
+
+  [405, 429, 503].forEach(statusCode =>
+    it(`errors with a status code ${statusCode} not indicated by the v1 spec are 500s`, () =>
```
Comment (adunkman) [inline web-api/src/v2/v2ApiWrapper.test.js]:
```suggestion
    it(`errors with a status code ${statusCode} not indicated by the v2 spec are 500s`, () =>
```

**Review by adunkman — COMMENTED** [review]
Looks like there are a few introduced linter errors in this pull request — I submitted suggested changes to fix them below, along with two questions.

------

In addition to the stories mentioned in the description, this PR contains: 

- A followup to https://github.com/ustaxcourt/ef-cms/pull/1552 in commit 44b27b4a2a883d7e514d826b856ea5246f7fd64d which removes the initial configuration of `DESTINATION_TABLE` and `MIGRATE_FLAG`.
- https://github.com/flexion/ef-cms/pull/8737
- [Devex 864: Maximum file size for unit tests](https://trello.com/c/TG3luUmS/864-maximum-file-size-for-unit-tests)
   - https://github.com/flexion/ef-cms/pull/8671
   - https://github.com/flexion/ef-cms/pull/8675
   - https://github.com/flexion/ef-cms/pull/8686
   - https://github.com/flexion/ef-cms/pull/8729
   - https://github.com/flexion/ef-cms/pull/8735
   - https://github.com/flexion/ef-cms/pull/8738
   - https://github.com/flexion/ef-cms/pull/8739
   - https://github.com/flexion/ef-cms/pull/8750
   - https://github.com/flexion/ef-cms/pull/8752
- [Devex 971: Boost test coverage to 99% on shared](https://trello.com/c/lYoJDA3y/971-boost-test-coverage-to-99-on-shared)
   - https://github.com/flexion/ef-cms/pull/8677
   - https://github.com/flexion/ef-cms/pull/8742
- [Devex 973: Boost test coverage to 99% on web-api](https://trello.com/c/5DkJY41g/973-boost-test-coverage-to-99-on-web-api)
   - https://github.com/flexion/ef-cms/pull/8709
- [Devex 335: Change camelCased folders to be consistent on web-client](https://trello.com/c/0RcWpGyM/335-change-camelcased-folders-to-be-consistent-on-web-client)
   - https://github.com/flexion/ef-cms/pull/8740

⚠️ Merging this pull request will require running the manual environment-specific terraform step. 

After this pull request is merged, we can update CircleCI configuration through the UI to remove the following variables: 
  - `SONAR_ORG`
  - `SHARED_SONAR_KEY`
  - `SHARED_SONAR_TOKEN`
  - `API_SONAR_KEY`
  - `API_SONAR_TOKEN`
  - `UI_SONAR_KEY`
  - `UI_SONAR_TOKEN`

File: web-api/migration-terraform/main/lambdas/migration-segments.js (hunk 5 above)
Comment (ericsorenson) [reply web-api/migration-terraform/main/lambdas/migration-segments.js:174]: Those check that the migrations have been run in the past, I don't think there's any harm in keeping them there.

File: shared/jest.config.js (hunk 1 above)
Comment (ericsorenson) [reply shared/jest.config.js]: Reverted, that devex item hasn't been entirely completed yet!

File: web-api/migration-terraform/main/lambdas/migration-segments.js (hunk 5 above)
Comment (adunkman) [reply web-api/migration-terraform/main/lambdas/migration-segments.js:174]: If we started up a new environment today, it wouldn’t have these migrations ran — and since the migration code has no capability to run these migrations, I don’t believe the code would be deployable. 

Am I missing something? I might be. 🤔

Comment (ericsorenson) [reply web-api/migration-terraform/main/lambdas/migration-segments.js:174]: Updated!

File: web-api/migration-terraform/main/lambdas/migration-segments.js [hunk 7]
```diff
@@ -41,8 +44,8 @@ const migrateRecords = async ({
   items,
   ranMigrations = {},
 }) => {
-  if (!ranMigrations['bug-0035-private-practitioner-representing.js']) {
-    applicationContext.logger.info('about to run bug migration 0035');
+  if (!ranMigrations['bugMigration0035.js']) {
```
Comment (adunkman) [inline web-api/migration-terraform/main/lambdas/migration-segments.js]: I believe this should be testing for `bug-0035-private-practitioner-representing.js` — that’s what this line was before this PR, and I believe your intention was to revert the changes. 

Could you say more about why this should be `bugMigration0035.js` instead?

File: web-api/migration-terraform/main/lambdas/migration-segments.js [hunk 8]
```diff
@@ -51,16 +54,21 @@ const migrateRecords = async ({
     items = await bugMigration0036(items);
   }
 
-  if (!ranMigrations['0036-phone-number-format.js']) {
+  if (!ranMigrations['migration0036.js']) {
```
Comment (adunkman) [inline web-api/migration-terraform/main/lambdas/migration-segments.js]: Same as above — I was expecting this to be reverted to `0036-phone-number-format.js`.

File: web-api/migration-terraform/main/lambdas/migration-segments.js [hunk 9]
```diff
@@ -51,16 +54,21 @@ const migrateRecords = async ({
...
     applicationContext.logger.debug('about to run migration 0036');
-    items = await migration0036(items);
+    items = await migration0036(items, documentClient);
```
Comment (adunkman) [inline web-api/migration-terraform/main/lambdas/migration-segments.js]: Looks like this is also an unintentional change — I don’t believe this migration takes a second argument.

File: web-api/migration-terraform/main/lambdas/migrations/0035-in-care-of-to-additional-name.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
+const createApplicationContext = require('../../../../src/applicationContext');
```
Comment (adunkman) [inline web-api/migration-terraform/main/lambdas/migrations/0035-in-care-of-to-additional-name.js]: This looks like a new migration introduced in this PR — and it doesn’t appear to be called from `migration-segments`.

File: web-api/migration-terraform/main/lambdas/migrations/0035-in-care-of-to-additional-name.test.js [hunk 11]
```diff
@@ -0,0 +1,123 @@
+const {
```
Comment (adunkman) [inline web-api/migration-terraform/main/lambdas/migrations/0035-in-care-of-to-additional-name.test.js]: Same for this file — I wasn’t expecting this migration (or it’s test) to be introduced here.

**Review by adunkman — CHANGES_REQUESTED** [review]
I believe there are a few bugs introduced in the revert commit, https://github.com/ustaxcourt/ef-cms/pull/1547/commits/0f9f8316826a5e6ed9531c9d501f7edbdba7cfda — let me know if it’s my misunderstanding!

File: web-api/migration-terraform/main/lambdas/migration-segments.js (hunk 5 above)
Comment (adunkman) [reply web-api/migration-terraform/main/lambdas/migration-segments.js:174]: Just circling back here to summarize for the future — we discussed this after standup this morning with @mmarcotte and @ericsorenson. 

Meta-level decisions:

- If migrations are removed, they should be deleted from the codebase with its tests, as well as these lines which call `hasMigrationRan`. Removing the `hasMigrationRan` calls at the same time as the migrations keeps the application deployable.
- Since the Court’s environments receive migrations after Flexion’s, @mmarcotte is responsible for removing outdated migrations going forward, and he doesn’t mind being nudged to do so if they’re getting in the way. 
- Removing migration code reduces clutter in this folder and keeps tests running fast. 

For this pull request:

- This PR should be updated to revert removing the other migrations, leaving them unchanged in the PR. 
- The PR should continue to include the introduction of the new migration needed.

**Review by adunkman — APPROVED** [review]
Looks great — thanks for taking a second look!
